### PR TITLE
UI Improvements: Modernized Scrollbars and Dark Mode Bug Fix

### PR DESCRIPTION
**Pull request by Tajmirul**
### Summary

- More modern and user-friendly scrollbars.
- In dark mode, there was a bug in the banner on the home page. `data-theme` was hard-coded to `light`, as a result, the search box was white in dark mode. Now it's also dynamic, coming from the Redux store.

By implementing these changes, we hope to make our software project more accessible and enjoyable to use for all users. We appreciate your time and consideration in reviewing this pull request and welcome any feedback you may have.

### Screenshots

| Before | After |
| ------ | ------ |
| image | image |
| image | image |

### CODE REVIEW

File: src/components/HomePage/HomePageHero.tsx
Comment (ahmedriad1) [inline src/components/HomePage/HomePageHero.tsx:1]: Please revert the changes in this file. The hero section was intentionally made to look like this.

File: src/components/HomePage/HomePageHero.module.scss
Comment (ahmedriad1) [inline src/components/HomePage/HomePageHero.module.scss:1]: Please revert the changes in this file. The hero section was intentionally made to look like this.

**Review by ahmedriad1 — CHANGES_REQUESTED** [review]
Salam alykum @Tajmirul,

JAK for your amazing contribution!

PR looks amazing masha'Allah, just left a couple of comments.

File: src/components/HomePage/HomePageHero.module.scss
Comment (Tajmirul) [reply src/components/HomePage/HomePageHero.module.scss:1]: Resolved

**Review by Tajmirul — COMMENTED** [review]
the hero section is in the previous style

**Review by osamasayed — APPROVED** [review]

**Comment by osamasayed** [comment]
Jazak Allahu Khairan @Tajmirul for your contribution. PR will be included in the upcoming release inshAllah.

**Comment by Tajmirul** [comment]
Wa antum fa-jazakumullahu khayran @osamasayed. I am happy to contribute to the project and I am glad that my PR will be included in the upcoming release. I look forward to seeing the project grow and succeed.

**Comment by Amandeep-Chauhan** [comment]
jai shree raaaaaam

**Comment by Amandeep-Chauhan** [comment]
modi modi modi

**Pull request merged**